FutureNorthants/northants-website
# Allow mark document as archived

Add a checkbox to Document media type to specify that the file should be marked as archived. I added it to the source document, rather than in the slice, as this should mean it only has to be marked as archived in one place in the CMS and it will update everywhere it is used. 

> Please note that if you have one link to a URL with 'rel=archived' and then another link to the same URL, without 'rel=archived', the URL will be assessed - you need to ensure all links mark it as 'archived'.

https://sitemorse.com/blog-article/9392/tech-update-archiving-old-content-cannot-be-updated/

## To Test
- `./drush.wrapper @docker config:import`
- `./drush.wrapper @docker cache:rebuild`
- Go to `/admin/content/media` in the CMS, find a document, then edit it and check the `Mark as archived` checkbox
- Edit a service page and add a File Download slice and select the document you have marked as archived. Also add a document that is not archived. Publish and save the page. 
- Use the below query to test that the archived document has archived: true and the other document is false

```graphql
{
  route(path: "your-test-page") {
    node {
      title
      body {
        value
        embeds {
          id
          paragraph {
            ... on FileDownloadParagraph{
              id
              files {
                title
                url
                type
                size
                id
                archived
              }
            }
          }
        }
      }
    }
  }
}
```
